src/indev/lv_indev.c

```diff
@@ -344,11 +347,26 @@ void lv_indev_reset(lv_indev_t * indev, lv_obj_t * obj)
                 indev->pointer.last_pressed = NULL;
             }
             if(obj == NULL || indev->pointer.act_obj == obj) {
-                indev->pointer.act_obj = NULL;
+                if(indev->pointer.act_obj) {
+                    /* Avoid recursive calls */
+                    act_obj = indev->pointer.act_obj;
+                    indev->pointer.act_obj = NULL;
+                    lv_obj_send_event(act_obj, LV_EVENT_INDEV_RESET, indev);
+                    act_obj = NULL;
+                }
             }
             if(obj == NULL || indev->pointer.last_obj == obj) {
                 indev->pointer.last_obj = NULL;
             }
+            if(obj == NULL || indev->pointer.scroll_obj == obj) {
+                if(indev->pointer.scroll_obj) {
+                    /* Avoid recursive calls */
+                    scroll_obj = indev->pointer.scroll_obj;
+                    indev->pointer.scroll_obj = NULL;
+                    lv_obj_send_event(scroll_obj, LV_EVENT_INDEV_RESET, indev);
+                    scroll_obj = NULL;
+                }
+            }
         }
```

Comment:
The code in `if(indev) {...}` is repeated in the `else {...}` branch as well. Could you move these section into a function like `indev_reset_core()`?

Reply:
Okey, I add a new function `indev_reset_core ` to reset indev.